src/sample.py

```diff
@@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
 		return True
 debugger = Debugger()
 hb.buffer_set_message_func (buf, debugger.message, 1, 0)
-hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
+hb.buffer_add_codepoints (buf, codepoints, 0, -1)
```

Comment:
Would it work if I just changed the function to add_utf32? and still use
ord() ? I don’t understand unicode encoding that well sorry

On Fri, Jun 17, 2016 at 2:30 PM, Khaled Hosny <notifications@github.com>
wrote:

> In src/sample.py
> <https://github.com/behdad/harfbuzz/pull/269#discussion_r67555245>:
>
> > @@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
> >  		return True
> >  debugger = Debugger()
> >  hb.buffer_set_message_func (buf, debugger.message, 1, 0)
> > -hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
> > +hb.buffer_add_codepoints (buf, codepoints, 0, -1)
>
> It accepts surrogate characters for one, which are invalid in UTF-32:
>
> a = "\uD800\uDFFF"
>
> This is a valid string in Python, but not valid UTF-32 string and if you
> pass such code units to _add_utf32() it will replace them with the
> replacement character but _add_codepoints() won’t.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/behdad/harfbuzz/pull/269/files/0a9c03bc45e82c95377431576ddc92e8cd32ddb4#r67555245>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACcEOvPW5yIbNA13B-ENuXNE56RLxGLpks5qMufEgaJpZM4I3_8R>
> .
>